lib/while.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file  Given a condition and function, continuously call the promisified version of that function
+ * sequentially and return the result once the exiting condition is met.
+ * @copyright Copyright (c) 2019 Olono, Inc.
+ */
+
+'use strict';
+
+/**
+ * Given a condition and function, continuously call the promisified version of that function sequentially
+ * and return the result once the exiting condition is met.
+ *
+ * The `condition` can access either the parent scoped variables or the results of `fn` which are passed in
+ * as the only parameter.
+ *
+ * @module while
+ * @example
+ * const the = require('await-the');
+ * let sum = 0;
+ * const condition = previousResult => sum < 10;
+ * const asyncFn = x => {
+ *     sum += x;
+ *     return sum * 10;
+ * }
+ * const result = await the.while(condition, asyncFn, 2);
+ * // will loop while sum < 10, then return the final function result
+ * // sum === 10
+ * // result === 100
+ *
+ * @param {Function} condition The condition to continue looping.
+ * @param {Function|Array} fn The function to be resolved (or rejected) every loop.
+ * @param {...*} args Variadic arguments to send to the function.
+ * @returns {*} The thrown error or the result.
+ */
+module.exports = async (condition, fn, ...args) => {
+    if ((await condition()) !== true) {
+        return;
+    }
+
+    const loop = async () => {
+        const result = await fn(...args);
+        if ((await condition(result)) === true) {
```

Comment:
does this need to `=== true` like regaular whiles aren't strict to true just truthy so 
```suggestion
        if ((await condition(result))) {
```

Reply:
Is that not what you meant in this comment?

> also probably want to support `condition === true` for infinite loops

https://github.com/olono/await-the/pull/31#discussion_r291341561